packages/gasket-request/lib/wrappers.js

```diff
@@ -0,0 +1,26 @@
+import { makeGasketRequest } from './request.js';
+
+/** @type {typeof import('./index.d.ts').withGasketRequestCache} */
+export function withGasketRequestCache(actionFn) {
+  const cache = new WeakMap();
+
+  return async (gasket, req, ...args) => {
+    const gasketRequest = await makeGasketRequest(req);
+
+    if (cache.has(gasketRequest)) {
+      return cache.get(gasketRequest);
+    }
+
+    const promise = actionFn(gasket, gasketRequest, ...args);
+    cache.set(gasketRequest, promise);
+    return await promise;
```

Comment:
Same feedback here about keeping Promises in cache. In my experience this can cause more memory retention issues, and also this doesn't take care of failed Promises being cached. Granted for the memory retention issue it's not as big of a deal since requests are short lived, but this should at least slim down the memory consumption when processing large amounts of requests concurrently.

```suggestion
    try {
      const result = await promise;
      cache.set(gasketRequest, result);
      return result;
    } catch (err) {
      cache.delete(gasketRequest);
      throw err;
    }
```